x-pack/plugins/observability_solution/observability_ai_assistant/server/rule_connector/index.ts

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { filter, lastValueFrom } from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { KibanaRequest, Logger } from '@kbn/core/server';
+import { AlertingConnectorFeatureId } from '@kbn/actions-plugin/common/connector_feature_config';
+import type {
+  ActionType as ConnectorType,
+  ActionTypeExecutorOptions as ConnectorTypeExecutorOptions,
+  ActionTypeExecutorResult as ConnectorTypeExecutorResult,
+} from '@kbn/actions-plugin/server/types';
+import { ConnectorAdapter } from '@kbn/alerting-plugin/server';
+import { SlackApiParamsSchema } from '@kbn/stack-connectors-plugin/common';
+import {
+  EmailParamsSchema,
+  JiraParamsSchema,
+  PagerdutyParamsSchema,
+  SlackParamsSchema,
+  WebhookParamsSchema,
+} from '@kbn/stack-connectors-plugin/server';
+import { ObservabilityAIAssistantRouteHandlerResources } from '../routes/types';
+import {
+  ChatCompletionChunkEvent,
+  MessageRole,
+  StreamingChatResponseEventType,
+} from '../../common';
+import { OBSERVABILITY_AI_ASSISTANT_CONNECTOR_ID } from '../../common/rule_connector';
+import { concatenateChatCompletionChunks } from '../../common/utils/concatenate_chat_completion_chunks';
+import { ChatFunctionClient } from '../service/chat_function_client';
+import { convertSchemaToOpenApi } from './convert_schema_to_open_api';
+import { CompatibleJSONSchema } from '../../common/functions/types';
+
+const connectorParamsSchemas: Record<string, CompatibleJSONSchema> = {
+  '.slack_api': convertSchemaToOpenApi(SlackApiParamsSchema),
```

Comment:
my bad, I think this is actually `.slack-api`

Reply:
I'll update

Reply:
No need it's [.slack_api](https://github.com/elastic/kibana/blob/main/x-pack/plugins/stack_connectors/common/slack_api/constants.ts#L8)

Reply:
thanks for double-checking!